### ISSUE TITLE
Add Alliance/War Feature

### ISSUE THREAD

**Issue by SatoGojo** [elementum-games/shinobi-chronicles#57]
Alliance with other villages, able to go into allied village and purchase ramen, spar, ect

**Comment by ViscaBarre** [comment]
Adding:

This could be tied to World Boss: Tailed Beasts (yet to be written) and their spawn condition.

Once enough blood has been spillt over time (several wars!), a tailed beast emerges. Add some conditions that trigger the spawning of specific beast out of many.

**Comment by ViscaBarre** [comment]
**Yasutake** has given his suggestions for the open war feature questions:


**Who decide to go into war?**
- The Kage, with agreement of 2/4 clan elders from the top 4 clans. Or 2 clan heads.


**How is the war begun?**
- Step by step. -> Skirmishes in areas that do not belong to either nation, possibly invoking other nations to defend or ban those villages from their boarders so they do not affect the local population, 
- After Skirmishes it leads for claims of territory, so a specific set of land, lets say 5 blocks horizontally and vertically (can be changed) 
- From the most territory claimed, it can then lead to seighes on the villages. In the end, the village with the most Territory, (will connect to points) wins, but kills also contribute to points. 
- To claim an area you need to either declare the area a contested territory and send Shinobi to capture it and fortify it, or (Refer to how genin can help) 
 

**Any requirements? Costs?** 

- A general agreement from the Chuunin+ Population that war needs to happen, which means somewhat a vote, or clan members putting pressure on their clanheads/clean elders (given they're top 4) to approach the Kage with the concept
Announcements? 
-> A declaration of war, banning the enemy targets from their lands, therefore making it illegal for them to enter, warranting the use of lethal force. 


**Who is the war declared on? How many targets?**

- A village can declare war on 2 targets, however if war is forced on the village from 4 ends, then it'll be over incredibly quickly in this idea.


**What are the motivators to go into war?**

- Mainly when a village is targeted by another, or one village wants the current funds of said village, thus they will attempt to defeat them in a war to sanction them, and in a sense, receive funds for the damages, and lives lost. 


**What effect does the war have on the normal gameplay of genin, chuunin, jounin and later sannin? Latter two aren't in yet, picture it instead.**

- Genin aren't allowed to travel into certain areas too close to the enemy territory 
- Chuunin are advised to supervise Genin Travel and support the frontlines. Travel is also marked unsafe
- Jounin will also watch the travel of Genin and Chuunin, leading to most likely a 3 man team if that's the case. Jounin will be directly on the frontliness, unless they're a medic.


**How will PVP change?**

- Being captured, Being interrogated, Being allowed to flee.

**When does the war end?**

- The war ends when the most points are won from territory or kills. Even both.


**What are the effects on the winning side? What about losing side?**

- The winning side secures a fraction (large or medium) of the losing villages funds as reoperations, and are banned from the claimed territory for a week or two. The winning side also now has that claimed territory to defend in other wars. As well as a training boost
- The losing side face the loss of funds, and can't travel to certain areas due to their loss. 
 

**How soon can war be declared again?**

- Give it a month cooldown. 

**What features are necessary to start a war?**

- A threshold of kills on your shinobi from a certain village (As Hokage) and the approval of Clan elders/Leaders


**What features would make the war interesting?**

- Capture, interrogation, injury putting the losing participant out of battle for a certain amount of time.

**In what ways can players participate and assist in wartime efforts?**

- Preferably even genin. Genin can fortify locations, and bases on their own. They can travel outward to secure a location or outpost, which troops can rally up at before taking off to another location.
- Genin can also help fortify village defenses, which are the stat required to be broken to fully sleigh a village and attack players


**What are the features players would be excited about when it comes to war?**

- The equality of opportunity, so it isn't so one sided one village just doesn't steam roll another. Meaning they'd have to be in a similar point ra

**Comment by ViscaBarre** [comment]
Discord discussion on prisoner of war aspect:

**Chi:** One concern that comes to mind is what will the captured people do during their capture: it'd very likely hinder their training. I do like the out of the box idea though.

**Yasutake:** Most likely, don't stop them from training while captured. They just can't leave. Like uncle Iroh getting swole in prison, if they can break out or not is ?

**Lone:** I don't think players should be kept hostage unless there is someway to break out eventually, as if the requirement is like you said to have the Kage send a diplomat to released PoWs, whos to say you don't have a bum kage who just doesn't care, or one that doesnt log in for 2-3 days, are you just stuck hostage for however long?

If there is someway to get free I'm not entirely against the idea but I don't think players should just be told to stop playing the game pretty much until someone else logs in to save them.

**Yasutake:** Breaking out would be ideal yeah, but if you have a bum kage who doesn't care, they lose influence for every day their POW is not secured and returned home. This is to deter Kages from not caring, as influence directly affects a Kage's time in office. 

If you won't save anyone, no one will fight for you, so you need to be proactive in getting them back, if they're too weak to get out themselves. 
You need to defeat two prison guards to escape.

**Yasutake:** Yeah, the systems have to work together, so breakouts need to be implemented eitherway to make it fun and fully functional on both sides.

**Lone:** Yeah I agree there

**Yasutake:** This is why i didnt really include heavy restrictions like training loss on the losing side, as then it's just telling the lower levels to screw off.

**Lone:** Yeah im not against the whole PoW idea.
**Lone:** Just let the hostage atleast have options or SOMETHING to do in the meanwhile

TL;DR: Implement "breaking out option", then PoW is gucci.
